src/Accelerators/NNPA/Transform/ZHigh/ZHighConstPropagation.cpp

```diff
@@ -91,44 +95,54 @@ ZHighStickifiedConstantOp createConstantForStick(PatternRewriter &rewriter,
     Value replacingValue, Value input, StringAttr layout) {
   Location loc = replacingValue.getLoc();
   Operation *op = input.getDefiningOp();
-  ArrayRef<int64_t> shape = mlir::cast<ShapedType>(input.getType()).getShape();
-  Type elementType = mlir::cast<ShapedType>(input.getType()).getElementType();
-  int rank = shape.size();
-
   // Read dense attributes.
   DenseElementsAttr dataAttr = mlir::dyn_cast_or_null<mlir::DenseElementsAttr>(
       op->getAttrOfType<::mlir::Attribute>("value"));
   assert(dataAttr && "Attribute is null");
-  // Read attributes's raw data.
-  std::vector<char> rawData;
-  getRawData(dataAttr, rawData);
-  // assert((rawData.size() == (uint64_t)getMemRefSizeInBytes(input)) &&
-  //        "Data size mismatched");
-
-  // Call stickify.
-  zdnn_tensor_desc pre_tfrmd_desc, tfrmd_desc;
-  // pre-transformed desc.
-  zdnn_data_layouts zDNNLayout =
-      convertLayoutAttrToZDNNDataLayout(rank, layout);
-  // If zDNNLayout is NHWC, we stickify directly from NCHW.
-  if (zDNNLayout == ZDNN_NHWC)
-    zDNNLayout = ZDNN_NCHW;
-  zdnn_data_types zDNNType = mlirTypeToZDNNType(elementType);
-  set_info_pre_transformed_desc(&pre_tfrmd_desc, zDNNLayout, zDNNType, shape);
-  // transformed desc.
-  zdnn_status status = generate_transformed_desc(&pre_tfrmd_desc, &tfrmd_desc);
-  assert(status == ZDNN_OK);
-  // Stick data using the software stickify.
-  zdnn_ztensor ztensor;
-  init_ztensor(&pre_tfrmd_desc, &tfrmd_desc, &ztensor);
-  status = allochelper_ztensor_alloc(&ztensor);
-  assert(status == ZDNN_OK);
-  status = stickify(&ztensor, rawData.data());
-  assert(status == ZDNN_OK);
-  // Emit a constant global in ZHigh dialect.
-  ZHighStickifiedConstantOp constantOp = emitZHighStickifiedConstant(
-      rewriter, loc, &ztensor, replacingValue.getType());
-
+  ZHighStickifiedConstantOp constantOp;
+  if (storeConstantsToFile && nnpaDelayStickifiedConstGen) {
+    constantOp = rewriter.create<ZHighStickifiedConstantOp>(loc,
+        replacingValue.getType(),
+        /*value=*/dataAttr,
+        /*stickified=*/rewriter.getBoolAttr(false),
+        /*allzero=*/nullptr,
+        /*alignment=*/rewriter.getI64IntegerAttr(4096));
+  } else {
+    ArrayRef<int64_t> shape =
```

Comment:
Stickify code here is the same as the code in `getBuffer()` for `ZLowStickifiedConstant`. Could you create a share function for it? Thanks!